### ISSUE TITLE
Is pause/resume working?

### ISSUE THREAD

**Issue by mark-hahn** [mikeal/morestreams#2]
I'm getting an error on the line `this.pause();` in the main.js code below, saying that the object has no attribute pause.  Pause isn't defined in main.js, is it supposed to be defined in `Stream`?

```
BufferedStream.prototype.write = function (chunk) {
  if (this.dest) {
    this.emit('data', chunk);
    return;
  }
  this.chunks.push(chunk);
  this.size += chunk.length;
  if (this.limit < this.size) {
    this.pause();
  }
}
```

I have been using the code for some time. This must have been the first time my buffer hit the limit.

I'm using 0.4.9 and my code is ...

```
server = http.createServer (req, res) ->
    req.pipe (reqBody = new BufferedStream)

    ...code snipped including async call ...

    reqBody.pipe http.request options, (resProx) ->
        res.writeHeader resProx.statusCode, resProx.headers
        resProx.pipe res
```

Has anyone tested the buffer hitting the limit?

**Comment by mikeal** [comment]
node core's pipe() function should add a pause and resume method.

**Comment by mark-hahn** [comment]
Any idea why it didn't?

**Comment by mikeal** [comment]
you know what, it's only added to the source stream on pipe, not the destination. i'll get this fixed today.

**Comment by mikeal** [comment]
this should be fixed now in master. once you verify i'll push a new release.

**Comment by mark-hahn** [comment]
Thanks.  It appears to work fine.  I set a limit of 10 bytes and put in a console.log that showed the pause happened.  I also tested it with a big limit and saw it work without the pause.  I have no way of knowing whether it actually paused but I'll trust you on that.

I'm surprised more people aren't using this.  I had a hell of a time chasing the problem down when I lost events.

May I suggest the docs for events mention this problem? I noticed the http.request docs mention you won't lose events when you follow the suggested pattern, it would be nice to say when you will lose events.
